### ISSUE TITLE
[Bug]: Issues with reset codes

### ISSUE THREAD

**Issue by MarketingPip** [MarketingPipeline/ctree.js#1]
There is a current issue with reset codes - not allowing things like `chalk` to apply background colors properly to this.

As shown as below - 

![image](https://github.com/user-attachments/assets/1a88fca6-29f5-4268-b29a-95ca0ba3c643)

Mentioning the wizard @jcubic - to see if he can possibly look where the ANSI reset codes need placed to properly achieve this. As well in hopes he see's this / might want to grab this library.

**Comment by jcubic** [comment]
I think that you can't change this behavior in `ctree`, I think that this is `chalk` issue. Unless you want to remove all ANSI colors when used with pipe. This is how ls/grep and other commands work by default. You can force colors with `--color` flag.

**Comment by MarketingPip** [comment]
@jcubic - you can (but it's a matter of properly applying the reset codes). Which I can not seem to figure out currently. And just figured your experience with escape sequences you might be able to see what's going wrong.

See [here](https://github.com/jcubic/isomorphic-lolcat/pull/1#issuecomment-2395669526) - as this is related to lolcat. (It appears I have a working formatted for piping colors from chalk bg colors etc - ie first match of BG code) correctly and returning a ANSI based result for terminal. (Stripping fg colors).

**Comment by jcubic** [comment]
I don't think that it will work, but you try to reset the color just after you use it. If you use it once for one character, reset it after that character.

But I think that it will work with chalk, it's really simple tool just add color at the beginning of the string. TO make it work you will need to use ANSI parser inside chalk, and it's not as trivial as adding color, I don't think that chalk is doing this, but I may be wrong.

**Comment by MarketingPip** [comment]
@jcubic - when I completely remove the reset code in "ctree.js" - this is the results.

![image](https://github.com/user-attachments/assets/93a3150c-5495-42c6-9e93-62ed09e94682)

- which it messes those two lines (and not sure how to properly format the reset codes to solve this). 

(but as fair as I know - we should be able to pipe things like these to other things).

**Comment by jcubic** [comment]
Sorry, no idea how to fix this. I think that when you have Unix tools, only the last in pipe should change color.

**Comment by MarketingPip** [comment]
@jcubic - correct - the last pipe is the final results with applied colors! But no worries - thank you for looking regardless!

**Comment by jcubic** [comment]
You can try to do what most tools are using, strip colors when using pipe.

**Comment by MarketingPip** [comment]
@jcubic - I don't want to do that. I want to apply colors with a pipe sequence ;)

(I got lolcat working this way - as said so I can "pipe" into it) a lolcat modified string and apply my chalk BG to it without removing it. (Keeping both the foreground colors applied and background).

**Comment by MarketingPip** [comment]
Noting here for self - the issue is due to [this](https://github.com/MarketingPipeline/ctree.js/blob/main/src/ctree.js#L71) line of code. Applying resets are not needed for "/" etc (xmas text).

As well reset codes are not needed to be applied for most / all methods.


This should solve not being able to properly add bg codes to example:

![image](https://github.com/user-attachments/assets/0f235b39-a33e-4891-b78d-7905907b43e1)

**Comment by MarketingPip** [comment]
Fixed in https://github.com/MarketingPipeline/ctree.js/commit/8a55c4b726e47294284e16bde7318768fbba0f67. Closes #1
